### ISSUE TITLE
[BUG] Invalid Operator

### ISSUE THREAD

**Issue by vukasinpetrovic** [fingers10/JqueryDataTablesServerSide#60]
**Describe the bug**
I tried using your methods for searching and sorting server functionality. Sorting works all right, but I get exception for searching method which says "Invalid Operator 'Explanation'", where Explanation is class property. I compared to your demo project but I cannot find difference nor what causes problem.
Here below are code snippets.

Stack trace
```
System.ArgumentException: Invalid Operator 'Explanation'.
   at JqueryDataTables.ServerSide.AspNetCoreWeb.Providers.DefaultSearchExpressionProvider.GetComparison(MemberExpression left, String op, Expression right)
   at JqueryDataTables.ServerSide.AspNetCoreWeb.Infrastructure.SearchOptionsProcessor`2.GetComparisonExpression(SearchTerm term, ParameterExpression obj)
   at JqueryDataTables.ServerSide.AspNetCoreWeb.Infrastructure.SearchOptionsProcessor`2.Apply(IQueryable`1 query, IEnumerable`1 columns)
   at Tamber.Repositories.PostRepository.GetReportsPaged(JqueryDataTablesParameters parameters) in /Users/vpetrovic/Documents/Projects/Web/Tamber/Tamber/Repositories/PostRepository.cs:line 443
   at Tamber.Services.PostService.GetPagedReportsForAdminPanel(JqueryDataTablesParameters param) in /Users/vpetrovic/Documents/Projects/Web/Tamber/Tamber/Services/PostService.cs:line 592
   at Tamber.Areas.AdminPanel.Controllers.PostController.GetPagedReports(JqueryDataTablesParameters param) in /Users/vpetrovic/Documents/Projects/Web/Tamber/Tamber/Areas/AdminPanel/Controllers/PostController.cs:line 47
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfIActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeInnerFilterAsync>g__Awaited|13_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResourceFilter>g__Awaited|24_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResourceExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Microsoft.AspNetCore.Diagnostics.EntityFrameworkCore.MigrationsEndPointMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Diagnostics.EntityFrameworkCore.DatabaseErrorPageMiddleware.Invoke(HttpContext httpContext)
   at Microsoft.AspNetCore.Diagnostics.EntityFrameworkCore.DatabaseErrorPageMiddleware.Invoke(HttpContext httpContext)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
```

This is a repository method where I call two methods for sorting and searching
```c#
        public async Task<List<PostReport>> GetReportsPaged(JqueryDataTablesParameters parameters)
        {
            IQueryable<PostReport> query = _context.PostReports
                .Include(pr => pr.ReportedBy)
                .Include(pr => pr.Post)
                .ThenInclude(p => p.PostedBy);
            
            query = SearchOptionsProcessor<PostReportDTO, PostReport>.Apply(query, parameters.Columns);
            query = SortOptionsProcessor<PostReportDTO, PostReport>.Apply(query, parameters);
                
            return await query
                .Skip((parameters.Start / parameters.Length) * parameters.Length)
                .Take(parameters.Length)
                .ToListAsync();
        }
```

This is my PostReport Model
```c#
    public class PostReport
    {
        [Key]
        public long Id { get; set; }
        
        public DateTime DateTime { get; set; }
        
        public long PostId { get; set; }
        
        [ForeignKey(name: "PostId")]
        public Post Post { get; set; }
        
        public string ReportedById { get; set; }
        
        [ForeignKey(name: "ReportedById")]
        public ApplicationUser ReportedBy { get; set; }

        public string Explanation { get; set; }
        
        public bool Resolved { get; set; }
    }
```

This is PostReportDTO class
```c#
    public class PostReportDTO
    {
        public long Id { get; set; }
        
        [Searchable]
        [Sortable]
        public DateTime DateTime { get; set; }
        
        public ElementDto ReportedBy { get; set; }

        public PostDTO Post { get; set; }
        
        [Searchable]
        [Sortable]
        public string Explanation { get; set; }
        
        public bool Resolved { get; set; }
    }
```

**Desktop (please complete the following information):**
 - OS: [e.g. iOS] MacOS Big Sur
 - Browser [e.g. chrome, safari] Chrome
 - Version [e.g. 22] 91
 - .NET Core 3.1

**Comment by fingers10** [comment]
@vukasinpetrovic try adding `[SearchableString]` attribute to your `Explanation` property in `PostReportDTO`

And what is the operator that you are sending fr JavaScript configuration? Please compare your jQuery Datatables initialisation script with my example. I'm sure you must be passing wrong operator from your script

**Comment by vukasinpetrovic** [comment]
@fingers10 I finally figured out what was the problem. I did not realise easily from documentation that I had to specify comparator string inside name property of columns, inside JS datatable initialisation code (image below). Also I added specific Searchable annotations (SearchableString, SearchableDate, etc.), but main problem was that comparator string.

![image](https://user-images.githubusercontent.com/23005422/120938541-85524700-c713-11eb-9d8a-ed059ed01001.png)
